tasks/rh-sign-cosign-image/tests/test-rh-sign-cosign-image-multiple-components.yaml

```diff
@@ -0,0 +1,122 @@
+              _TEST_MANIFEST_LIST_MDIGEST2="sha256:1111-2"
+              _TEST_MANIFEST_LIST_MDIGEST3="sha256:1111-3"
+              _TEST_MANIFEST_LIST_REFERENCE="quay.io/test-product----test-image0@${_TEST_MANIFEST_LIST_DIGEST}"
+              _TEST_MANIFEST_LIST_REPO="registry.redhat.io/test-product/test-image0"
+
+              _TEST_MANIFEST_LIST_OCI_DIGEST="sha256:0000"
+              _TEST_MANIFEST_LIST_OCI_MDIGEST1="sha256:0000-1"
+              _TEST_MANIFEST_LIST_OCI_MDIGEST2="sha256:0000-2"
+              _TEST_MANIFEST_LIST_OCI_MDIGEST3="sha256:0000-3"
+              _TEST_MANIFEST_LIST_OCI_REFERENCE="quay.io/test-product----test-image0@${_TEST_MANIFEST_LIST_OCI_DIGEST}"
+              _TEST_MANIFEST_LIST_OCI_REPO="registry.redhat.io/test-product/test-image0"
+
+              cat > $(workspaces.data.path)/snapshot_spec.json << EOF
+              {
+                "application": "myapp",
+                "components": [
+                  {
+                    "name": "comp0",
+                    "containerImage": "$_TEST_MANIFEST_LIST_REFERENCE",
+                    "repository": "$_TEST_MANIFEST_LIST_REPO"
+                  },
+                  {
+                    "name": "comp0",
+                    "containerImage": "$_TEST_MANIFEST_LIST_OCI_REFERENCE",
+                    "repository": "$_TEST_MANIFEST_LIST_OCI_REPO"
+                  }
+                ]
+              }
+              EOF
+
+              cat > $(workspaces.data.path)/data.json << EOF
+              {
+                "sign": {
+                  "request": "hacbs-signing-pipeline",
+                  "configMapName": "signing-config-map"
+                }
+              }
+              EOF
+    - name: run-task
+      taskRef:
+        name: rh-sign-cosign-image
+      params:
+        - name: snapshotPath
+          value: snapshot_spec.json
+        - name: dataPath
+          value: data.json
+        - name: secretName
+          value: 'test-rh-sign-cosign-image-secret'
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      runAfter:
+        - setup
+    - name: check-result
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      taskSpec:
+        steps:
+          - name: check-result
+            image: quay.io/konflux-ci/release-service-utils:38c3bfd7479c86b832cba5b61f9bbde40c469393
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+              echo "check results"
+              _TEST_PUB_REPO="registry.redhat.io/test-product/test-image0"
+              _TEST_REPO="quay.io/test-product----test-image0"
+
+              EXPECTED=$(cat <<EOF
+              inspect --raw docker://${_TEST_REPO}@sha256:1111
+              inspect --raw docker://${_TEST_REPO}@sha256:0000
+              EOF
+              )
+              CALLS=$(cat "$(workspaces.data.path)/mock_skopeo_calls")
+              [ "$CALLS" = "$EXPECTED" ] || exit 1
```

Comment:
The `|| exit 1` is redundant here. Also, I would suggest running it as `test "$CALLS" = "$EXPECTED"`. It does the same thing, but it makes it consistent with our other tests and it makes the tests easy to spot in the check-results script.

Reply:
Yeah, I usually do && || so I put it there just as a habit. I'll change it to test

Reply:
fixed

Reply:
Cool, thanks.